### PR TITLE
Use sync.Map for podCacheInterface

### PR DESCRIPTION
**Pull request by qinqon**
**What this PR does / why we need it**:
Currenctly all the VMIs are serialized at a mutex once per vmi+interface
at virt-handler, this change uses a sync.Map that will not serialize on
disjoint keys.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by davidvossel — APPROVED** [review]
/lgtm
/approve

**Comment by qinqon** [comment]
@davidvossel added the sync.Map for the phase1 ran map.
/hold cancel

**Comment by qinqon** [comment]
@davidvossel everything is fine now you can LGTM it.

**Review by rmohr — COMMENTED** [review]
Hm, having all this type casting directly in the critical code paths and not factored out is not something I see as a real improvement. :/

Do you think you can wrap the sync maps in type-safe structs? The `ok` from the conversasion in these structs can directly be catched and we can panic since that would be obvious programming errors which can not really happen with a clean interface.

**Comment by qinqon** [comment]
@rmohr is ready for review

File: pkg/virt-handler/vm.go [hunk 1]
```diff
@@ -1848,20 +1818,16 @@ func (d *VirtualMachineController) closeLauncherClient(vmi *v1.VirtualMachineIns
...
-	// maps require locks for concurrent access
-	d.launcherClientLock.Lock()
-	defer d.launcherClientLock.Unlock()
-
-	clientInfo, ok := d.launcherClients[vmi.UID]
-	if ok && clientInfo.client != nil {
-		if clientInfo.client != nil {
-			clientInfo.client.Close()
-			close(clientInfo.domainPipeStopChan)
+	clientInfo, loaded := d.launcherClients.LoadAndDelete(vmi.UID)
```
Comment (rmohr) [inline pkg/virt-handler/vm.go:1821]: I don't think that we can do the get and delete at the same time. We do some cleanup which has to be done. IF we would have an error with e.g. `os.RemoveAll` we would not clean up the ghost record. I think you have to get here and do the delete where it was done before.

**Review by rmohr — CHANGES_REQUESTED** [review]
One severe thing which needs to be addressed. Whenever I see a wrapped sync.Map in use I am uncertain if it is worth it, or if we should just use a normal map with a simple mutex in the wrapper.

**Comment by qinqon** [comment]
/hold

There are some concerns

File: pkg/virt-handler/vm.go (hunk 1 above)
Comment (qinqon) [reply pkg/virt-handler/vm.go]: Going back to Load/Delete that the original code, nice catch btw.

**Comment by qinqon** [comment]
/hold

Fixed comments

**Comment by rmohr** [comment]
/lgtm
/hold

@davidvossel can you have a look too? I want a second eye on it.

**Review by davidvossel — APPROVED** [review]
/hold cancel
/lgtm
/approve

**Pull request merged**